### PR TITLE
Rename to HedgeDoc

### PR DESCRIPTION
**Pull request by davidmehren**
Fixes part of #452

### CODE REVIEW

**Comment by davidmehren** [comment]
I *think* that should be all for now. Things I didn't touch:
- Release Notes
- Links in setup guides not maintained by us
- History
- Etherpad migration guide

File: docs/guides/auth/github.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Authentication guide - GitHub
 
-*Note:* This guide was written before the renaming. Just replace `HackMD` with `CodiMD` in your mind :smile: thanks!
+*Note:* This guide was written before the renaming. Just replace `HackMD` with `HedgeDoc` in your mind :smile: thanks!
```
Comment (SISheogorath) [inline docs/guides/auth/github.md:3]: Maybe we should extend this statement :D

**Comment by ericgaspar** [comment]
Shall I push a PR to master for YunoHost setup doc?

**Comment by davidmehren** [comment]
> Shall I push a PR to master for YunoHost setup doc?

You can of course prepare a PR, but there's no need to hurry as we need to [find a logo](https://community.codimd.org/t/time-to-find-the-hedgedoc-logo/171) before we can actually rename and move the repos.

**Comment by mrdrogdrog** [comment]
I rebased this branch on master and added the new logo.

**Comment by mrdrogdrog** [comment]
Many links were changed in this PR. Therefore we need to adjust some stuff.

- [] Subdomains on hedgedoc.org (and hedgedoc.net? maybe we could just create a wildcard cname entry for hedgedoc.net?) @SISheogorath @ErikMichelson 
  - [] community.hedgedoc.org
  - [] demo.hedgedoc.org
  - [] social.hedgedoc.org
  - [] translate.hedgedoc.org
- [] Move github repositories
  - [] github.com/hedgedoc/server
  - [] github.com/hedgedoc/cli
  - [] github.com/hedgedoc/container
  - [] github.com/hedgedoc/meta-marked
  - [] github.com/hedgedoc/js-sequence-diagrams
- [] Change names in github.com/hedgedoc/cli
- [] Inform other projects about the renaming 
  - [] https://github.com/linuxserver/docker-codimd/
  - [] YunoHost

**Comment by mrdrogdrog** [comment]
I merged the todo list with the one in https://github.com/codimd/server/issues/452

File: LICENSE [hunk 2]
```diff
@@ -629,12 +629,12 @@ to attach them to the start of each source file to most effectively
 state the exclusion of warranty; and each file should have at least
 the "copyright" line and a pointer to where the full notice is found.
 
-    CodiMD - Realtime collaborative markdown notes on all platforms.
+    HedgeDoc - Realtime collaborative markdown notes on all platforms.
```
Comment (SISheogorath) [inline LICENSE:632]: Do we want to adjust the slogan?

File: docs/dev/getting-started.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 **Notice:** *There's [specialised instructions for docker](../setup/docker.md) or [heroku](../setup/heroku.md), if you prefer running code this way!*
 
-1. Clone the repository with `git clone https://github.com/codimd/server.git codimd-server`
+1. Clone the repository with `git clone https://github.com/hedgedoc/server.git hedgedoc-server`
```
Comment (SISheogorath) [inline docs/dev/getting-started.md:7]: This needs some adjustment.

File: docs/dev/openapi.yml [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
-  title: CodiMD
-  description: CodiMD is an open source collaborative note editor. Several tasks of CodiMD can be automated through this API.
+  title: HedgeDoc
+  description: HedgeDoc is an open source collaborative note editor. Several tasks of HedgeDoc can be automated through this API.
   version: 1.6.0
   contact:
-    name: CodiMD on GitHub
-    url: https://github.com/codimd/server
+    name: HedgeDoc on GitHub
+    url: https://github.com/hedgedoc/server
```
Comment (SISheogorath) [inline docs/dev/openapi.yml:9]: This needs some adjustment as well

File: docs/history.md [hunk 5]
```diff
@@ -34,4 +34,8 @@ move the repository, so we simply forked it. We still welcome the HackMD team
 as part of our community, especially since a large portion of this code base
 originated with them.
 
-*For the debate that lead to this step, please refer to the [governance debate](https://github.com/hackmdio/hackmd/issues/1170) and [the announcement of the new repository](https://github.com/codimd/server/issues/10).*
+*For the debate that lead to this step, please refer to the [governance debate](https://github.com/hackmdio/hackmd/issues/1170) and [the announcement of the new repository](https://github.com/hedgedoc/server/issues/10).*
+
+## CodiMD became HedgeDoc
+
+<to be filled>
```
Comment (SISheogorath) [inline docs/history.md:41]: Please don't leave `TODO` in documents 😉

**Review by SISheogorath — CHANGES_REQUESTED** [review]
Just some minor things.

A TODO to solve, we need to update a few more URLs and I guess it's intentional that we don't change the environment variable prefix (which is mentioned in the rename issue.)

File: docs/dev/openapi.yml (hunk 4 above)
Comment (mrdrogdrog) [reply docs/dev/openapi.yml]: done

File: docs/dev/getting-started.md (hunk 3 above)
Comment (mrdrogdrog) [reply docs/dev/getting-started.md]: done

File: LICENSE (hunk 2 above)
Comment (mrdrogdrog) [reply LICENSE]: done

File: docs/history.md (hunk 5 above)
Comment (mrdrogdrog) [reply docs/history.md]: I added a draft.

File: locales/de.json [hunk 6]
```diff
@@ -118,5 +117,13 @@
...
     "Join the community": "Tritt der Community bei",
-    "Imprint": "Impressum"
+    "Imprint": "Impressum",
+    "changed": "changed",
+    "created": "created",
+    "Freely - Anyone can edit": "Freely - Anyone can edit",
+    "Editable - Signed-in people can edit": "Editable - Signed-in people can edit",
+    "Limited - Signed-in people can edit (forbid guests)": "Limited - Signed-in people can edit (forbid guests)",
+    "Locked - Only owner can edit": "Locked - Only owner can edit",
+    "Protected - Only owner can edit (forbid guests)": "Protected - Only owner can edit (forbid guests)",
```
Comment (SISheogorath) [inline locales/de.json:127]: This shouldn't be part of this PR.

File: docs/history.md [hunk 7]
```diff
@@ -34,4 +34,12 @@ move the repository, so we simply forked it. We still welcome the HackMD team
 as part of our community, especially since a large portion of this code base
 originated with them.
 
-*For the debate that lead to this step, please refer to the [governance debate](https://github.com/hackmdio/hackmd/issues/1170) and [the announcement of the new repository](https://github.com/codimd/server/issues/10).*
+*For the debate that lead to this step, please refer to the [governance debate](https://github.com/hackmdio/hackmd/issues/1170) and [the announcement of the new repository](https://github.com/hedgedoc/hedgedoc/issues/10).*
+
+## CodiMD became HedgeDoc
+
+Many people opened issues in the codimd server repository for bugs and features, which they saw at the HackMD codimd.
```
Comment (SISheogorath) [inline docs/history.md:41]: Maybe this sentence should start with something like "With two projects, both actively maintained, one by the HackMD team, and this hardfork by the community, confusion about the name and feature set became visible in form of issue reports about HackMD's version this repository, which as originally called `codimd/server`."

File: docs/history.md [hunk 8]
```diff
@@ -34,4 +34,12 @@ move the repository, so we simply forked it. We still welcome the HackMD team
...
+The fact, that there have been two projects with the same purpose and the same name was very confusing.
+The solution was to find a unique, new name. This process started in April 2020 with a [call to participation](https://community.codimd.org/t/renaming-yet-another-time/102).
+It took some time, but in [July 2020 the community agreed on the name "HedgeDoc"](https://community.codimd.org/t/codimd-becomes-hedgedoc/170) and some months later [a new logo was found](https://community.codimd.org/t/time-to-find-the-hedgedoc-logo/171/4).
+Finally, in November 2020 version 1.7 was released. Even if the renaming was originally planned for version 2.0, the dev team decided to release 1.7 under the new name to create a smooth transition.
```
Comment (SISheogorath) [inline docs/history.md:45]: Here I would replace dev with the full word development. I'm also not sure if the original plan vs actual plan is that important here. It might be more useful to tell about the 2.0 development that happened in parallel and the unexpected release of 1.7.0 even when the release notes of 1.6.0 stated something else.

**Review by SISheogorath — CHANGES_REQUESTED** [review]
Some more points to fix, but we are getting there! 👍🏻

**Comment by davidmehren** [comment]
@SISheogorath I'd actually remove the `history.md`, as we have the new landing page that explains the history. Maybe move some more text there, but having the history explainer at two places seems redundant.

**Comment by SISheogorath** [comment]
@davidmehren then we should drastically enrich the history section on the landing page, pointing to all links that we have in the `history.md`. But that's definitely a solution I can live with.

**Comment by SISheogorath** [comment]
Uhm, it seems like the history page is one giant SVG graphics. How does one maintain the texts in there?

**Comment by mrdrogdrog** [comment]
The history is an SVG because that was the fastest way to create a nice graph with correct text positions. It was orginally planned as "this-is-how-it-could-look-like". We plan to replace it with some css magic, later. 

So right now that's technically not possible to insert links :/

**Comment by mrdrogdrog** [comment]
So right now we still need the history.md, but we can move all the texts to this page as soon as I have time to unpack my CSS-Magicbox.

File: docs/guides/migrations-and-breaking-changes.md [hunk 9]
```diff
@@ -9,20 +9,20 @@ We dropped support for node 6 with this version. If you have any trouble running
...
 
-When you use our [container repository](https://github.com/codimd/container) 
-(which was previously `codimd-container`) all you can simply run `git pull` and
+When you use our [container repository](https://github.com/hedgedoc/container) 
+(which was previously `hedgedoc-container`) all you can simply run `git pull` and
 your `docker-compose.yml` will be updated.
 
 When you setup things yourself, make sure you use the new image:
-[`quay.io/codimd/server`](https://quay.io/repository/codimd/server?tab=tags).
+[`quay.io/hedgedoc/server`](https://quay.io/repository/hedgedoc/server?tab=tags).
```
Comment (DerMolly) [inline docs/guides/migrations-and-breaking-changes.md:25]:
```suggestion
[`quay.io/hedgedoc/hedgedoc`](https://quay.io/repository/hedgedoc/hedgedoc?tab=tags).
```
the repo now is called `hedgedoc/hedgedoc` and not `hedgedoc/server`

File: docs/setup/docker.md [hunk 10]
```diff
@@ -1,20 +1,20 @@
-# CodiMD Docker Image
+# HedgeDoc Docker Image
 
-[![Try in PWD](https://cdn.rawgit.com/play-with-docker/stacks/cff22438/assets/images/button.png)](http://play-with-docker.com?stack=https://github.com/codimd/container/raw/master/docker-compose.yml&stack_name=codimd)
+[![Try in PWD](https://cdn.rawgit.com/play-with-docker/stacks/cff22438/assets/images/button.png)](http://play-with-docker.com?stack=https://github.com/hedgedoc/container/raw/master/docker-compose.yml&stack_name=hedgedoc)
 
 ## Debian-based version
 
-[![Docker Repository on Quay](https://quay.io/repository/codimd/server/status "Docker Repository on Quay")](https://quay.io/repository/codimd/server)
+[![Docker Repository on Quay](https://quay.io/repository/hedgedoc/server/status "Docker Repository on Quay")](https://quay.io/repository/hedgedoc/server)
```
Comment (DerMolly) [inline docs/setup/docker.md:7]:
```suggestion
[![Docker Repository on Quay](https://quay.io/repository/hedgedoc/hedgedoc/status "Docker Repository on Quay")](https://quay.io/repository/hedgedoc/hedgedoc)
```

File: docs/setup/docker.md [hunk 11]
```diff
@@ -1,20 +1,20 @@
...
 
 ## Alpine-based version
 
-[![Docker Repository on Quay](https://quay.io/repository/codimd/server/status "Docker Repository on Quay")](https://quay.io/repository/codimd/server)
+[![Docker Repository on Quay](https://quay.io/repository/hedgedoc/server/status "Docker Repository on Quay")](https://quay.io/repository/hedgedoc/server)
```
Comment (DerMolly) [inline docs/setup/docker.md:11]:
```suggestion
[![Docker Repository on Quay](https://quay.io/repository/hedgedoc/hedgedoc/status "Docker Repository on Quay")](https://quay.io/repository/hedgedoc/hedgedoc)
```

File: docs/guides/auth/keycloak.md [hunk 12]
```diff
@@ -24,7 +24,7 @@ You may note that a separate realm is specified throughout this tutorial. It is
 1. Select Client Scopes from the sidebar, and begin to create a new client scope using the Create button.
 2. Ensure that the **Name** field is set to `id`.
 3. Create a new mapper under the Mappers tab. This should reference the User Property `id`. `Claim JSON Type` should be String and all switches below should be enabled. Save the mapper.
-4. Go to the client you set up in the previous steps using the Clients page, then choose the Client Scopes tab. Apply the scope you've created. This should mitigate errors as seen in [codimd/server#56](https://github.com/codimd/server/issues/56), as the `/userinfo` endpoint should now bring back the user's ID under the `id` key as well as `sub`.
+4. Go to the client you set up in the previous steps using the Clients page, then choose the Client Scopes tab. Apply the scope you've created. This should mitigate errors as seen in [hedgedoc/server#56](https://github.com/hedgedoc/hedgedoc/issues/56), as the `/userinfo` endpoint should now bring back the user's ID under the `id` key as well as `sub`.
```
Comment (DerMolly) [inline docs/guides/auth/keycloak.md:27]:
```suggestion
4. Go to the client you set up in the previous steps using the Clients page, then choose the Client Scopes tab. Apply the scope you've created. This should mitigate errors as seen in [hedgedoc/hedgedoc#56](https://github.com/hedgedoc/hedgedoc/issues/56), as the `/userinfo` endpoint should now bring back the user's ID under the `id` key as well as `sub`.
```

**Review by InnayTool — APPROVED** [review]

**Comment by DerMolly** [comment]
`/s/release-notes` seems to need upgrading, does it not?

**Comment by DerMolly** [comment]
image

**Comment by mrdrogdrog** [comment]
> image
> Do we want to remove that fork-awesome icon? It's not the icon anymore so it looks weird…

corrected

**Review by DerMolly — APPROVED** [review]

**Review by mrdrogdrog — APPROVED** [review]

**Review by SISheogorath — APPROVED** [review]

File: README.md [hunk 13]
```diff
@@ -1,21 +1,20 @@
-# CodiMD
+![HedgeDoc Logo](https://raw.githubusercontent.com/hedgedoc/hedgedoc-logo/main/LOGOTYPE/SVG/HedgeDoc-Logo%204b.svg)
```
Comment (ErikMichelson) [inline README.md:1]: The image is a bit too large for a README in my opinion.
Maybe add some size restrictions and center it horizontally.

File: README.md [hunk 14]
```diff
@@ -1,21 +1,20 @@
...
-**Note**: CodiMD is in the process of [changing its name to HedgeDoc](https://github.com/codimd/server/issues/452).
+HedgeDoc lets you create real-time collaborative markdown notes. You can test-drive
+it by visiting our [HedgeDoc demo server][hedgedoc-demo].
 
 It is inspired by Hackpad, Etherpad and similar collaborative editors. This
 project originated with the team at [HackMD](https://hackmd.io) and now forked
 into its own organisation. [A longer writeup can be read in the history doc](docs/history.md).
 
-[![CodiMD 1.3.2 with its feature demonstration page open](docs/images/CodiMD-1.3.2-features.png)][codimd-demo-features]
+[![HedgeDoc 1.3.2 with its feature demonstration page open](docs/images/HedgeDoc-1.7.0-features.png)][hedgedoc-demo-features]
```
Comment (ErikMichelson) [inline README.md:17]: In fact this screenshot shows version 1.7.0, not 1.3.2.

File: docs/configuration.md [hunk 15]
```diff
@@ -46,28 +46,28 @@ these are rarely used for various reasons.
...
 | `protocolUseSSL` | `CMD_PROTOCOL_USESSL` | **`false`** or `true`                                            | set to use SSL protocol for resources path (only applied when domain is set)                                      |
 | `useSSL`         |                       | **`false`** or `true`                                            | set to use SSL server (if `true`, will auto turn on `protocolUseSSL`)                                             |
 | `urlAddPort`     | `CMD_URL_ADDPORT`     | **`false`** or `true`                                            | set to add port on callback URL (ports `80` or `443` won't be applied) (only applied when domain is set)          |
-| `allowOrigin`    | `CMD_ALLOW_ORIGIN`    | **`['localhost']`**, `['codimd.org']`, `[localhost, codimd.org]` | domain name whitelist (use comma to separate)                                                                     |
+| `allowOrigin`    | `CMD_ALLOW_ORIGIN`    | **`['localhost']`**, `['hedgedoc.org']`, `[localhost, hedgedoc.org]` | domain name whitelist (use comma to separate)                                                                     |
```
Comment (ErikMichelson) [inline docs/configuration.md:70]:
```suggestion
| `allowOrigin`    | `CMD_ALLOW_ORIGIN`    | **`['localhost']`**, `['hedgedoc.org']`, `['localhost', 'hedgedoc.org']` | domain name whitelist (use comma to separate)                                                                     |
```

File: docs/guides/auth/github.md [hunk 16]
```diff
@@ -1,7 +1,5 @@
 # Authentication guide - GitHub
 
-*Note:* This guide was written before the renaming. Just replace `HackMD` with `CodiMD` in your mind :smile: thanks!
```
Comment (ErikMichelson) [inline docs/guides/auth/github.md:3]: Why is this statement removed and not 'CodiMD' replaced with "HedgeDoc"?

File: lib/config/index.js [hunk 17]
```diff
@@ -157,7 +157,7 @@ for (let i = keys.length; i--;) {
 
 // Notify users about the prefix change and inform them they use legacy prefix for environment variables
 if (Object.keys(process.env).toString().indexOf('HMD_') !== -1) {
-  logger.warn('Using legacy HMD prefix for environment variables. Please change your variables in future. For details see: https://github.com/codimd/server#environment-variables-will-overwrite-other-server-configs')
+  logger.warn('Using legacy HMD prefix for environment variables. Please change your variables in future. For details see: https://github.com/hedgedoc/hedgedoc#environment-variables-will-overwrite-other-server-configs')
```
Comment (ErikMichelson) [inline lib/config/index.js:160]: This link is dead, as the relevant section is not part of the README anymore.

File: public/icons/site.webmanifest [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+            "type": "image/png"
+        },
+        {
+            "src": "/icons/android-chrome-512x512.png",
+            "sizes": "512x512",
+            "type": "image/png"
+        }
+    ],
+    "theme_color": "#303030",
+    "background_color": "#303030"
```
Comment (ErikMichelson) [inline public/icons/site.webmanifest:17]: These are no theme colors of the HedgeDoc branding anymore.

File: public/views/hedgedoc.ejs [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+<html lang="en">
+
+<head>
+    <%- include hedgedoc/head %>
+</head>
+
+<body>
+    <%- include hedgedoc/header %>
+    <%- include hedgedoc/body %>
+    <%- include hedgedoc/footer %>
```
Comment (ErikMichelson) [inline public/views/hedgedoc.ejs:11]: Footer is an empty file that can be removed.

File: public/views/includes/favicon.ejs [hunk 20]
```diff
@@ -0,0 +1,11 @@
...
+<link rel="icon" type="image/png" sizes="32x32" href="<%- serverURL %>/icons/favicon-32x32.png">
+<link rel="icon" type="image/png" sizes="16x16" href="<%- serverURL %>/icons/favicon-16x16.png">
+<link rel="manifest" href="<%- serverURL %>/icons/site.webmanifest">
+<link rel="mask-icon" href="<%- serverURL %>/icons/safari-pinned-tab.svg" color="#b51f08">
+<link rel="shortcut icon" href="<%- serverURL %>/icons/favicon.ico">
+<meta name="apple-mobile-web-app-title" content="HedgeDoc - <%= __('Collaborative markdown notes') %>">
+<meta name="application-name" content="HedgeDoc - <%= __('Collaborative markdown notes') %>">
+<meta name="msapplication-TileColor" content="#b51f08">
+<meta name="msapplication-config" content="<%- serverURL %>/icons/browserconfig.xml">
+<meta name="theme-color" content="#303030">
```
Comment (ErikMichelson) [inline public/views/includes/favicon.ejs:11]: This theme color is not part of the HedgeDoc branding anymore.

File: public/views/index/body.ejs [hunk 21]
```diff
@@ -40,10 +40,13 @@
 
             <div id="home" class="section"<% if(signin) { %> style="display:none;"<% } %>>
                 <div class="inner cover">
-                    <h1 class="cover-heading"><i class="fa fa-file-text"></i> CodiMD</h1>
+                    <h1 class="cover-heading"><img src="<%- serverURL %>/banner/banner_vertical_color.svg" width="400" alt="HedgeDoc logo"></h1>
                     <p class="lead">
                         <%= __('Best way to write and share your knowledge in markdown.') %>
                     </p>
+                    <span class="alert alert-info" style="display: inline-block;">
+                        CodiMD was renamed to HedgeDoc. <a href="https://hedgedoc.github.io/history/" rel="noopener noreferrer" target="_blank" class="text-danger">Read more</a>
```
Comment (ErikMichelson) [inline public/views/index/body.ejs:48]: Do we want to point to `hedgedoc.github.io/history/` or to `hedgedoc.org/history/`?

File: public/views/shared/help-modal.ejs [hunk 22]
```diff
@@ -15,13 +15,13 @@
                                 <h3 class="panel-title"><%= __('Contacts') %></h3>
                             </div>
                             <div class="panel-body">
-                                <a href="https://community.codimd.org" target="_blank"><i class="fa fa-users fa-fw"></i> <%= __('Join the community') %></a>
+                                <a href="https://community.hedgedoc.org" target="_blank"><i class="fa fa-users fa-fw"></i> <%= __('Join the community') %></a>
                                 <br>
                                 <a href="https://matrix.to/#/#hedgedoc:matrix.org" target="_blank"><i class="fa fa-hashtag fa-fw"></i> <%= __('Meet us on %s', 'Matrix') %></a>
```
Comment (ErikMichelson) [inline public/views/shared/help-modal.ejs:20]: Do we wanto to use our new subdomain chat.hedgedoc.org here?

**Review by ErikMichelson — CHANGES_REQUESTED** [review]
Additional remarks:
- unneeded file: `public/icons/html_code.html`
- unneeded file: `public/views/hedgedoc/footer.ejs`

File: README.md (hunk 13 above)
Comment (mrdrogdrog) [reply README.md:1]: horizontal center is not possible with only markdown

File: docs/guides/auth/github.md (hunk 16 above)
Comment (mrdrogdrog) [reply docs/guides/auth/github.md:3]: Because the word "hackmd" was replaced with "hedgedoc"

Comment (ErikMichelson) [reply docs/guides/auth/github.md:3]: I don't see any updated screenshots for that file in this PR.

File: public/icons/site.webmanifest (hunk 18 above)
Comment (mrdrogdrog) [reply public/icons/site.webmanifest:17]: I changed the theme color, but not not the background color.

File: README.md (hunk 13 above)
Comment (ErikMichelson) [reply README.md:1]: That's true. I just checked how nextjs has done that in their README. Seems they just use an image with resolution of 1800x600.

File: lib/config/index.js (hunk 17 above)
Comment (mrdrogdrog) [reply lib/config/index.js:160]: I changed the link to the configuration doc

File: docs/guides/auth/github.md (hunk 16 above)
Comment (mrdrogdrog) [reply docs/guides/auth/github.md:3]: done

File: README.md (hunk 13 above)
Comment (mrdrogdrog) [reply README.md]: I added a resolution tag

Comment (mrdrogdrog) [reply README.md]: I replaced the image

**Review by ErikMichelson — APPROVED** [review]
